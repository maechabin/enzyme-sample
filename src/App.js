import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      text: '',
      inputValue: '',
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  handleChange(value) {
    this.setState({
      inputValue: value,
    });
  }

  handleClick() {
    this.setState({
      text: this.state.inputValue,
      inputValue: '',
    });
  }

  render() {
    return (
      <div className={this.state.text}>
        <Title text={this.state.text} />
        <Input handleChange={this.handleChange} value={this.state.inputValue} />
        <Button handleClick={this.handleClick} />
      </div>
    );
  }
}

export const Title = (props) => {
  return(
    <h1>Hello { props.text }</h1>
  );
}

export const Input = (props) => {
  function handleChange(event) {
    props.handleChange(event.target.value);
  }

  return (
    <input onChange={handleChange} value={props.value} />
  );
}

export const Button = (props) => {
  const handleClick = () => {
    props.handleClick();
  }
  return (
    <button onClick={handleClick}>送信</button>
  );
}
