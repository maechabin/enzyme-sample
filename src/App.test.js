import React from 'react';
import ReactDOM from 'react-dom';
import { App, Title, Input, Button } from './App';
import renderer from 'react-test-renderer';
import  { shallow, mount } from 'enzyme';

/**
 * Appコンポーネントのテスト
 */
describe('<App />', () => {
  const wrapper = shallow(<App />);

  test('子コンポーネントの存在確認', () => {
    expect(wrapper.find(Title).length).toBe(1);
    expect(wrapper.find(Input).length).toBe(1);
    expect(wrapper.find(Button).length).toBe(1);
  });

  test('setStateでthis.state.textを更新した時のclass名', () => {
    wrapper.setState({
      text: 'XXX',
    });
    expect(wrapper.find('.XXX').length).toBe(1);
  });

  test('handleChangeの呼び出し', () => {
    const setStateSpy = jest.spyOn(App.prototype, 'setState');
    wrapper.instance().handleChange('XXX');
    expect(setStateSpy).toHaveBeenCalledWith({
      inputValue: 'XXX',
    });
  });

  test('handleClickの呼び出し', () => {
    const setStateSpy = jest.spyOn(App.prototype, 'setState');
    wrapper.setState({
      text: 'XXX',
    });
    wrapper.instance().handleClick();
    expect(setStateSpy).toHaveBeenCalledWith({
      text: 'XXX',
      inputValue: '',
    });
  });

  test('スナップショット', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/**
 * Titleコンポーネントのテスト
 */
describe('<Title />', () => {
  test('h1要素の存在確認', () => {
    const wrapper = shallow(<Title text={'React'} />);
    expect(wrapper.find('h1').length).toBe(1);
  });

  test('propsの値', () => {
    const wrapper = shallow(<Title text={'React'} />);
    expect(wrapper.text()).toBe('Hello React');
    wrapper.setProps({ text: 'World' });
    expect(wrapper.text()).toBe('Hello World');
  });

  test('スナップショット', () => {
    const tree = renderer
      .create(<Title text={'React'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/**
 * Inputコンポーネントのテスト
 */
describe('<Input />', () => {
  test('input要素の存在確認', () => {
    const wrapper = shallow(<Input />);
    expect(wrapper.find('input').length).toBe(1);
  });

  test('changeイベント発火時のコールバック関数の呼び出し', () => {
    const handleChangeSpy = jest.fn();
    const wrapper = mount(<Input handleChange={handleChangeSpy} />);

    const event = { target: { value: 'aaa' } };
    wrapper.find('input').simulate('change', event);
    expect(handleChangeSpy).toHaveBeenCalledWith('aaa');
  });

  test('スナップショット', () => {
    const handleChangeSpy = jest.fn();
    const value = 'XXX';
    const tree = renderer
      .create(<Input onChange={handleChangeSpy} value={value} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

/**
 * Buttonコンポーネントのテスト
 */
describe('<Button />', () => {
  test('要素の存在', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.find('button').length).toBe(1);
  });
  
  test('clickイベント発火時のコールバック関数の呼び出し', () => {
    const handleClickSpy = jest.fn();
    const wrapper = shallow(<Button handleClick={handleClickSpy} />);
    wrapper.find('button').simulate('click');
    expect(handleClickSpy).toHaveBeenCalled();
  });

  test('スナップショット', () => {
    const handleClickSpy = jest.fn();
    const tree = renderer
      .create(<Button handleClick={handleClickSpy} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
