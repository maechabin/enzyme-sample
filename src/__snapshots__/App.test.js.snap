// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> スナップショット 1`] = `
<div
  className=""
>
  <h1>
    Hello 
    
  </h1>
  <input
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    送信
  </button>
</div>
`;

exports[`<Button /> スナップショット 1`] = `
<button
  onClick={[Function]}
>
  送信
</button>
`;

exports[`<Input /> スナップショット 1`] = `
<input
  onChange={[Function]}
  value="XXX"
/>
`;

exports[`<Title /> スナップショット 1`] = `
<h1>
  Hello 
  React
</h1>
`;
